app/src/components/Map/VesselInfoPanel.jsx

```diff
@@ -1,86 +1,151 @@
 import React, { Component } from 'react';
-import vesselPanel from '../../../styles/components/c-vessel-panel.scss';
+import classnames from 'classnames';
 import iso3311a2 from 'iso-3166-1-alpha-2';
 import { RadioGroup, Radio } from 'react-radio-group';
 
+import vesselPanelStyles from '../../../styles/components/c-vessel-info-panel.scss';
+
 class VesselInfoPanel extends Component {
 
   constructor(props) {
     super(props);
-    this.changeVesselTrackDisplayMode = this.props.changeVesselTrackDisplayMode.bind(this);
+
+    this.state = {
+      vesselInfo: this.props.vesselInfo,
+      vesselVisibility: this.props.vesselVisibility
+    };
   }
 
-  render() {
-    if (!this.props.vesselInfo || !Object.keys(this.props.vesselInfo).length) {
-      return null;
+  componentDidMount() {
+    this.setVars();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // new vessel selected
+    if (this.state.vesselInfo.callsign !== nextProps.vesselInfo.callsign) {
+      if (nextProps.vesselPosition) {
+        this.setVesselPosition(nextProps.vesselPosition);
+      }
     }
 
-    let marineTrafficLink = (
-      <a
-        href={`http://www.marinetraffic.com/en/ais/details/ships/mmsi:${this.props.vesselInfo.mmsi}`}
-        target="_blank"
-      >
-        here
-      </a>);
+    Object.assign(this.state, nextProps);
+    this.setState(this.state);
+  }
 
-    let flag = '';
-    if (this.props.vesselInfo.flag) {
-      flag = iso3311a2.getCountry(this.props.vesselInfo.flag);
-    }
+  setVars() {
+    this.el = document.querySelector('#vesselBox');
+  }
 
-    let extraInfo = [];
-    if (this.props.vesselInfo.latitude && this.props.vesselInfo.longitude) {
-      extraInfo.push(
-        <p key="vesselPanelLat">Lat:<span id="vesselPanelLat">{this.props.vesselInfo.latitude}</span></p>
-      );
-      extraInfo.push(
-        <p key="vesselPanelLong">Long: <span id="vesselPanelLong">{this.props.vesselInfo.longitude}</span></p>
-      );
-    }
-    if (this.props.vesselInfo.timestamp) {
-      let humanDate = new Date(this.props.vesselInfo.timestamp).toISOString().slice(0, 10);
-      extraInfo.push(
-        <p key="vesselPanelDate">Date: <span id="vesselPanelDate">{humanDate}</span></p>
-      );
+  setVesselPosition(position) {
+    const top = position.top;
+    const left = position.left;
+
+    this.el.setAttribute('style', `top: ${top + 30}px; left: ${left - 25}px`);
```

Comment:
Should be done in a render otherwise on the next render we could loose the position.